### ISSUE TITLE
Upgrade to a modern thrift

### ISSUE THREAD

**Issue by solongordon** [twitter/scrooge#85]
I'm attempting to use Scrooge to generate a Java Thrift client, but the generated code does not compile with Thrift 0.7.0 or higher. The generated Client class is declared:

```
public static class Client implements TServiceClient, Iface {
```

However, TServiceClient was changed from an interface to an abstract class in Thrift 0.7.0 so this does not compile.

Any thoughts? Thanks.

**Comment by bzz** [comment]
@sprsquish Is this still an issue: am I right that scrooge 3.17.0 with "experimental-java" still generates only libthrift 0.5 compatible code? 

If that's true, that pretty much might be worth documenting somewhere to save people's time who try to follow Thrift 0.9 http://thrift.apache.org/tutorial/java with scrooge

**Comment by eirslett** [comment]
experimental-java generates code for Thrift 0.9.
But it still doesn't cover all edge cases, like struct constants.

**Comment by adriancole** [comment]
Using the maven plugin, I can't seem to get a `$FinagleClient` subclass to generate.

Has the 'experimental-java' language with `--finagle` option been tested, lately. Simply switching the language to scala with the following mojo generates the `$FinagleClient` fine in scala.

```xml
<plugin>
    <groupId>com.twitter</groupId>
    <artifactId>scrooge-maven-plugin</artifactId>
    <!-- 3.17.0 is busted per #184 -->
    <version>3.16.0</version>
    <configuration>
        <!-- experimental-java doesn't generate the $FinagleClient subclass -->
        <language>scala</language>
        <thriftOpts>
            <thriftOpt>--finagle</thriftOpt>
        </thriftOpts>
    </configuration>
    <executions>
        <execution>
            <id>thrift-sources</id>
            <phase>generate-sources</phase>
            <goals>
                <goal>compile</goal>
            </goals>
        </execution>
    </executions>
</plugin>
```

**Comment by diguage** [comment]
Setting <language>experimental-java</language>, it can generate 0.9-compatible files.

The document should indicate it.

**Comment by benjumanji** [comment]
@adriancole #182

**Comment by mosesn** [comment]
We would like to update to a modern thrift version, but it's going to be a larger effort internally.

**Comment by mosesn** [comment]
We've finally [scheduled][0] this.  Should hopefully happen in the next few months.

[0]: https://finagle.github.io/blog/2017/03/20/spring-planning/

**Comment by muuki88** [comment]
This was never an issue with Scala, right?

**Comment by yufangong** [comment]
Hi, we upgraded libthrift to 0.10.0, it should solve this issue. If that is not the case please open another issue. Thank you.

**Comment by muuki88** [comment]
Awesome 🤩

This will be included in the next 18.4 release?
And is it the standard libthrift or an adapted Twitter version?

**Comment by yufangong** [comment]
@muuki88  yeah, this already already merged as of in 997f2464ca04998ce5b7a73c56c7667d1754a01b. And this is standard libthrift, not a twitter forked one!
